pyanalyze/suggested_type.py

```diff
@@ -122,14 +131,19 @@ def display_suggested_type(value: Value) -> Tuple[str, Optional[Dict[str, Any]]]
             # exist, and we should be using a Callable type instead anyway.
             metadata = None
         else:
-            suggested_type = stringify_object(value.typ)
-            imports = []
-            if isinstance(value.typ, str):
-                if "." in value.typ:
-                    imports.append(value.typ)
-            elif safe_getattr(value.typ, "__module__", None) != "builtins":
-                imports.append(suggested_type.split(".")[0])
-            metadata = {"suggested_type": suggested_type, "imports": imports}
+            typ_str = stringify_object(value.typ)
+            typ_name = typ_str.split(".")[-1]
+            scope_value = scopes.get(typ_name, node, VisitorState.check_names)
```

Comment:
I'm a bit nervous about this because in FunctionScope calls to `.get()` have side effects. I think this would be safer (note the docstring of `StackedScopes.get` explains that None is allowed): 

```suggestion
            scope_value = scopes.get(typ_name, None, VisitorState.check_names)
```

Also worth noting that this will behave differently for parameter and return types. For parameter types, this code is executed after the module has been fully processed, which means that all scopes except the module and function scope have been popped off again. But for return types, it's executed during type checking, so we can see any enclosing class and function scopes. That introduces some potential for edge cases, but it's probably fine since this is just a suggestion.